sql: v22.1.9: unhydrated enum in ShowHistogram

This issue was autofiled by Sentry. It represents a crash or reported error on a live cluster with telemetry enabled.

Sentry link: https://cockroach-labs.sentry.io/issues/4069613456/?referrer=webhooks_plugin

Panic message:

>types.go:2541: use of enum metadata before hydration as an enum: × ×
(1) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:761
  | runtime.gopanic
  | 	GOROOT/src/runtime/panic.go:1038
  | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
  | runtime.gopanic
  | 	GOROOT/src/runtime/panic.go:1038
  | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).ensureHydratedEnum
  | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2541
  | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).EnumGetIdxOfPhysical
  | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2506
  | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep
  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:4542
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside.Decode
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside/decode.go:267
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).EnsureDecoded
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:251
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).stringWithAlloc
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:66
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).String
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:75
  | github.com/cockroachdb/cockroach/pkg/sql.(*planner).ShowHistogram.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/show_histogram.go:87
  | github.com/cockroachdb/cockroach/pkg/sql.(*delayedNode).startExec
  | 	github.com/cockroachdb/cockroach/pkg/sql/delayed.go:50
  | github.com/cockroachdb/cockroach/pkg/sql.startExec.func2
  | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:516
  | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112
  | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297
  | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79
  | github.com/cockroachdb/cockroach/pkg/sql.walkPlan
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
  | github.com/cockroachdb/cockroach/pkg/sql.startExec
  | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:519
  | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start
  | 	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146
  | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:120
  | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
  | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92
  | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start
  | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:119
  | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run
  | 	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:753
  | github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run
  | 	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:469
  | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run
  | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:303
  | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
  | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:597
  | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
  | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1445
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1506
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1163
  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:699
Wraps: (2) while executing: SHOW HISTOGRAM 854593616702341122
Wraps: (3) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:761
  | runtime.gopanic
  | 	GOROOT/src/runtime/panic.go:1038
  | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
  | runtime.gopanic
  | 	GOROOT/src/runtime/panic.go:1038
  | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).ensureHydratedEnum
  | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2541
  | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).EnumGetIdxOfPhysical
  | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2506
  | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep
  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:4542
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside.Decode
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside/decode.go:267
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).EnsureDecoded
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:251
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).stringWithAlloc
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:66
  | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).String
  | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:75
  | github.com/cockroachdb/cockroach/pkg/sql.(*planner).ShowHistogram.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/show_histogram.go:87
  | github.com/cockroachdb/cockroach/pkg/sql.(*delayedNode).startExec
  | 	github.com/cockroachdb/cockroach/pkg/sql/delayed.go:50
  | github.com/cockroachdb/cockroach/pkg/sql.startExec.func2
  | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:516
  | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112
  | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297
  | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79
  | github.com/cockroachdb/cockroach/pkg/sql.walkPlan
  | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
  | github.com/cockroachdb/cockroach/pkg/sql.startExec
  | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:519
  | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start
  | 	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146
  | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:120
  | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
  | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92
  | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start
  | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:119
  | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run
  | 	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:753
  | github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run
  | 	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:469
  | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run
  | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:303
  | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
  | 	github.com/cockr...


<details>
<summary>Stacktrace (expand for inline code snippets):</summary>

https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/conn_executor.go#L760-L762 in pkg/sql.(*Server).ServeConn.func1
GOROOT/src/runtime/panic.go#L1037-L1039 in runtime.gopanic
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/colexecerror/error.go#L57-L59 in pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
GOROOT/src/runtime/panic.go#L1037-L1039 in runtime.gopanic
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/types/pkg/sql/types/types.go#L2540-L2542 in pkg/sql/types.(*T).ensureHydratedEnum
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/types/pkg/sql/types/types.go#L2505-L2507 in pkg/sql/types.(*T).EnumGetIdxOfPhysical
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/sem/tree/datum.go#L4541-L4543 in pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/rowenc/keyside/decode.go#L266-L268 in pkg/sql/rowenc/keyside.Decode
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/rowenc/encoded_datum.go#L250-L252 in pkg/sql/rowenc.(*EncDatum).EnsureDecoded
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/rowenc/encoded_datum.go#L65-L67 in pkg/sql/rowenc.(*EncDatum).stringWithAlloc
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/rowenc/encoded_datum.go#L74-L76 in pkg/sql/rowenc.(*EncDatum).String
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/show_histogram.go#L86-L88 in pkg/sql.(*planner).ShowHistogram.func1
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/delayed.go#L49-L51 in pkg/sql.(*delayedNode).startExec
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/plan.go#L515-L517 in pkg/sql.startExec.func2
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/walk.go#L111-L113 in pkg/sql.(*planVisitor).visitInternal.func1
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/walk.go#L296-L298 in pkg/sql.(*planVisitor).visitInternal
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/walk.go#L78-L80 in pkg/sql.(*planVisitor).visit
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/walk.go#L42-L44 in pkg/sql.walkPlan
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/plan.go#L518-L520 in pkg/sql.startExec
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/plan_node_to_row_source.go#L145-L147 in pkg/sql.(*planNodeToRowSource).Start
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/colflow/flow_coordinator.go#L119-L121 in pkg/sql/colflow.(*FlowCoordinator).Start.func1
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/colexecerror/error.go#L91-L93 in pkg/sql/colexecerror.CatchVectorizedRuntimeError
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/colflow/flow_coordinator.go#L118-L120 in pkg/sql/colflow.(*FlowCoordinator).Start
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/execinfra/processorsbase.go#L752-L754 in pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/flowinfra/flow.go#L468-L470 in pkg/sql/flowinfra.(*FlowBase).Run
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/colflow/vectorized_flow.go#L302-L304 in pkg/sql/colflow.(*vectorizedFlow).Run
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/distsql_running.go#L596-L598 in pkg/sql.(*DistSQLPlanner).Run
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/distsql_running.go#L1444-L1446 in pkg/sql.(*DistSQLPlanner).PlanAndRun
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/conn_executor_exec.go#L1505-L1507 in pkg/sql.(*connExecutor).execWithDistSQLEngine
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/conn_executor_exec.go#L1162-L1164 in pkg/sql.(*connExecutor).dispatchToExecutionEngine
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/conn_executor_exec.go#L698-L700 in pkg/sql.(*connExecutor).execStmtInOpenState
https://github.com/cockroachdb/cockroach/blob/e438c2f89282e607e0e6ca1d38b2e0a622f94493/pkg/sql/conn_executor_exec.go#L143-L145 in pkg/sql.(*connExecutor).execStmt
</details>

```
pkg/sql/conn_executor.go in pkg/sql.(*Server).ServeConn.func1 at line 761
GOROOT/src/runtime/panic.go in runtime.gopanic at line 1038
pkg/sql/colexecerror/error.go in pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1 at line 58
GOROOT/src/runtime/panic.go in runtime.gopanic at line 1038
pkg/sql/types/pkg/sql/types/types.go in pkg/sql/types.(*T).ensureHydratedEnum at line 2541
pkg/sql/types/pkg/sql/types/types.go in pkg/sql/types.(*T).EnumGetIdxOfPhysical at line 2506
pkg/sql/sem/tree/datum.go in pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep at line 4542
pkg/sql/rowenc/keyside/decode.go in pkg/sql/rowenc/keyside.Decode at line 267
pkg/sql/rowenc/encoded_datum.go in pkg/sql/rowenc.(*EncDatum).EnsureDecoded at line 251
pkg/sql/rowenc/encoded_datum.go in pkg/sql/rowenc.(*EncDatum).stringWithAlloc at line 66
pkg/sql/rowenc/encoded_datum.go in pkg/sql/rowenc.(*EncDatum).String at line 75
pkg/sql/show_histogram.go in pkg/sql.(*planner).ShowHistogram.func1 at line 87
pkg/sql/delayed.go in pkg/sql.(*delayedNode).startExec at line 50
pkg/sql/plan.go in pkg/sql.startExec.func2 at line 516
pkg/sql/walk.go in pkg/sql.(*planVisitor).visitInternal.func1 at line 112
pkg/sql/walk.go in pkg/sql.(*planVisitor).visitInternal at line 297
pkg/sql/walk.go in pkg/sql.(*planVisitor).visit at line 79
pkg/sql/walk.go in pkg/sql.walkPlan at line 43
pkg/sql/plan.go in pkg/sql.startExec at line 519
pkg/sql/plan_node_to_row_source.go in pkg/sql.(*planNodeToRowSource).Start at line 146
pkg/sql/colflow/flow_coordinator.go in pkg/sql/colflow.(*FlowCoordinator).Start.func1 at line 120
pkg/sql/colexecerror/error.go in pkg/sql/colexecerror.CatchVectorizedRuntimeError at line 92
pkg/sql/colflow/flow_coordinator.go in pkg/sql/colflow.(*FlowCoordinator).Start at line 119
pkg/sql/execinfra/processorsbase.go in pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run at line 753
pkg/sql/flowinfra/flow.go in pkg/sql/flowinfra.(*FlowBase).Run at line 469
pkg/sql/colflow/vectorized_flow.go in pkg/sql/colflow.(*vectorizedFlow).Run at line 303
pkg/sql/distsql_running.go in pkg/sql.(*DistSQLPlanner).Run at line 597
pkg/sql/distsql_running.go in pkg/sql.(*DistSQLPlanner).PlanAndRun at line 1445
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).execWithDistSQLEngine at line 1506
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).dispatchToExecutionEngine at line 1163
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).execStmtInOpenState at line 699
pkg/sql/conn_executor_exec.go in pkg/sql.(*connExecutor).execStmt at line 144
```

| Tag | Value |
|---|---|
| Cockroach Release | `v22.1.9` |
| Cockroach SHA: | e438c2f89282e607e0e6ca1d38b2e0a622f94493 |
| Platform | linux amd64 |
| Distribution | CCL |
| Environment | v22.1.9 |
| Command | server |
| Go Version | ``|
| # of CPUs ||
| # of Goroutines ||


Jira issue: CRDB-26694

I can reproduce this on v22.1.9  with `cockroach demo`:
```
CREATE TYPE enum1 as ENUM ('hello', 'hi');
CREATE TABLE mytab (x int, y enum1);
INSERT INTO mytab VALUES (1, 'hello'), (2, 'hello'), (3, 'hi');
ANALYZE mytab;
SHOW STATISTICS FOR TABLE mytab;
```
This will print something like the following:
```
  statistics_name | column_names |          created           | row_count | distinct_count | null_count | avg_size |    histogram_id
------------------+--------------+----------------------------+-----------+----------------+------------+----------+---------------------
  NULL            | {rowid}      | 2023-05-05 22:19:48.667957 |         3 |              3 |          0 |        9 | 862633291511136257
  NULL            | {x}          | 2023-05-05 22:19:48.667957 |         3 |              3 |          0 |        2 | 862633291518672897
  NULL            | {y}          | 2023-05-05 22:19:48.667957 |         3 |              2 |          0 |        3 | 862633291523784705
```
Now run:
```
SHOW HISTOGRAM 862633291523784705;
```
(using the value of `histogram_id` for column `y`). This causes the system to crash:
```
*
* ERROR: a SQL panic has occurred while executing the following statement:
* SHOW HISTOGRAM 862633484993232897
*
*
* ERROR: a panic has occurred!
* use of enum metadata before hydration as an enum: unknown_enum 0xc004fd0000
* (1) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:761
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:1038
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:1038
*   | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).ensureHydratedEnum
*   | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2541
*   | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).EnumGetIdxOfPhysical
*   | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2506
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:4539
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside.Decode
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside/decode.go:267
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).EnsureDecoded
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:251
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).stringWithAlloc
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:66
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).String
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:75
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planner).ShowHistogram.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/show_histogram.go:87
*   | github.com/cockroachdb/cockroach/pkg/sql.(*delayedNode).startExec
*   | 	github.com/cockroachdb/cockroach/pkg/sql/delayed.go:50
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec.func2
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:516
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79
*   | github.com/cockroachdb/cockroach/pkg/sql.walkPlan
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:519
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:120
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:119
*   | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:753
*   | github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:469
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:303
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:597
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1445
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1506
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1163
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:699
* Wraps: (2) while executing: SHOW HISTOGRAM 862633484993232897
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:761
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:1038
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:1038
*   | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).ensureHydratedEnum
*   | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2541
*   | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).EnumGetIdxOfPhysical
*   | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2506
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:4539
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside.Decode
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside/decode.go:267
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).EnsureDecoded
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:251
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).stringWithAlloc
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:66
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).String
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:75
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planner).ShowHistogram.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/show_histogram.go:87
*   | github.com/cockroachdb/cockroach/pkg/sql.(*delayedNode).startExec
*   | 	github.com/cockroachdb/cockroach/pkg/sql/delayed.go:50
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec.func2
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:516
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79
*   | github.com/cockroachdb/cockroach/pkg/sql.walkPlan
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:519
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:120
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:119
*   | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:753
*   | github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:469
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:303
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:597
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1445
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1506
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1163
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:699
* Wraps: (4) assertion failure
* Wraps: (5) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).ensureHydratedEnum
*   | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2541
*   | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).EnumGetIdxOfPhysical
*   | 	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2506
*   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep
*   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:4539
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside.Decode
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside/decode.go:267
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).EnsureDecoded
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:251
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).stringWithAlloc
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:66
*   | github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).String
*   | 	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:75
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planner).ShowHistogram.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/show_histogram.go:87
*   | github.com/cockroachdb/cockroach/pkg/sql.(*delayedNode).startExec
*   | 	github.com/cockroachdb/cockroach/pkg/sql/delayed.go:50
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec.func2
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:516
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79
*   | github.com/cockroachdb/cockroach/pkg/sql.walkPlan
*   | 	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan.go:519
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start
*   | 	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:120
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:119
*   | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:753
*   | github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:469
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:303
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:597
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
*   | 	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1445
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1506
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1163
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:699
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:144
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1891
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1895
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1817
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn
*   | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:763
* Wraps: (6) use of enum metadata before hydration as an enum: unknown_enum 0xc004fd0000
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *withstack.withStack (4) *assert.withAssertionFailure (5) *withstack.withStack (6) *errutil.leafError
*
*
* ERROR: Queued as error 6fac4d1af7ba47118864d4d0dbd03865
*
panic: use of enum metadata before hydration as an enum: unknown_enum 0xc004fd0000 [recovered]
	panic: use of enum metadata before hydration as an enum: unknown_enum 0xc004fd0000 [recovered]
	panic: use of enum metadata before hydration as an enum: unknown_enum 0xc004fd0000

goroutine 3485 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc0011be000, {0x9e7acd0, 0xc00417e980}, {0x851caa0, 0xc006354030})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1062 +0x1b4
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1()
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:761 +0x53
panic({0x851caa0, 0xc006354030})
	GOROOT/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x3f6
panic({0x851caa0, 0xc006354030})
	GOROOT/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).ensureHydratedEnum(...)
	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2541
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).EnumGetIdxOfPhysical(0x0, {0xc00459a051, 0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/sql/types/pkg/sql/types/types.go:2506 +0x3ee
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.GetEnumComponentsFromPhysicalRep(0xc004fd0000, {0xc00459a051, 0xc00480ee40, 0x5a68aa7b00000000})
	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:4539 +0x27
github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside.Decode(0xc0058cd430, 0xc004fd0000, {0xc00459a050, 0x6dd214e, 0xc00324c201}, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside/decode.go:267 +0x2417
github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).EnsureDecoded(0xc0058cd7f8, 0x15865878, 0x4bc3f09)
	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:251 +0x194
github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).stringWithAlloc(0xc0058cd7f8, 0x0, 0x4)
	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:66 +0x85
github.com/cockroachdb/cockroach/pkg/sql/rowenc.(*EncDatum).String(...)
	github.com/cockroachdb/cockroach/pkg/sql/rowenc/encoded_datum.go:75
github.com/cockroachdb/cockroach/pkg/sql.(*planner).ShowHistogram.func1({0x9e7acd0, 0xc005565880}, 0xc0011be6f0)
	github.com/cockroachdb/cockroach/pkg/sql/show_histogram.go:87 +0x4f2
github.com/cockroachdb/cockroach/pkg/sql.(*delayedNode).startExec(0xc00213c370, {{0x9e7acd0, 0xc005565880}, 0xc000c93180, 0xc0011be6f0})
	github.com/cockroachdb/cockroach/pkg/sql/delayed.go:50 +0x65
github.com/cockroachdb/cockroach/pkg/sql.startExec.func2({0xc0058cda70, 0x4010ad0}, {0x9e7c6d8, 0xc00213c370})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:516 +0x147
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112 +0x3e
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc0058cdb80, {0x9e7c6d8, 0xc00213c370}, {0x89b0b39, 0xd})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297 +0x14c9
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc0058cdb80, {0x9e7c6d8, 0xc00213c370})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79 +0xf7
github.com/cockroachdb/cockroach/pkg/sql.walkPlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
github.com/cockroachdb/cockroach/pkg/sql.startExec({{0x9e7acd0, 0xc005565880}, 0xc000c93180, 0xc0011be6f0}, {0x9e7c6d8, 0xc00213c370})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:519 +0x125
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start(0xc00219e700, {0x9e7acd0, 0xc005565880})
	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146 +0x87
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:120 +0x34
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x200000000)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start(0xc0028937a0, {0x9e7acd0, 0xc005565880})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:119 +0x6b
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc0028937a0, {0x9e7acd0, 0xc005565880})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:753 +0x3a
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc003b029c0, {0x9e7acd0, 0xc005565880}, 0xc0028937a0)
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:469 +0x258
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc0011be9e8, {0x9e7acd0, 0xc005565880}, 0xc0052c0000)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:303 +0x205
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00289c500, {0x9e7acd0, 0xc005565780}, 0xc0020aa000, 0xc0052c0000, 0xc00550d000, 0xc003e9e700, 0xc0011be9e8, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:597 +0xcb6
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0x9e7acd0, {0x9e7acd0, 0xc005565780}, 0xc0011be6f0, 0xc0020aa000, 0xc001f526c0, {{0x9e7c6d8, 0xc00213c370}, 0x0}, 0xc003e9e700)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1445 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc0011be000, {0x9e7acd0, 0xc005565780}, 0xc0011be6f0, 0xc005565780, {0x3e889480, 0xc001f526c0}, 0x0, 0xc00480ed78)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1506 +0x613
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc0011be000, {0x9e7acd0, 0xc005565780}, 0xc0011be6f0, {0x3e889480, 0xc001f526c0})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1163 +0xa2f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc0011be000, {0x9e7ad78, 0xc0055e4750}, {{0x9eabab8, 0xc002813820}, {0xc002ea3755, 0x21}, 0x0, 0x0}, 0x0, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:699 +0x2251
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc0011be000, {0x9e7ad78, 0xc0055e4750}, {{0x9eabab8, 0xc002813820}, {0xc002ea3755, 0x21}, 0x0, 0x0}, 0x0, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:144 +0x59e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1({{{0x9eabab8, 0xc002813820}, {0xc002ea3755, 0x21}, 0x0, 0x0}, {0x2b10fe70, 0xedbe7786b, 0x0}, {0x2b110258, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1891 +0x2f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc0011be000)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1895 +0xb08
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc0011be000, {0x9e7acd0, 0xc00417e980}, 0xc003f77dd0, {0x5400, 0x15000, 0xc0028a1b00, 0x0, 0x0}, 0xc00219ba60)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1817 +0x278
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0x9e7acd0, {0x9e7acd0, 0xc00417e980}, {0x3}, {0x5400, 0x15000, 0xc0028a1b00, 0x0, 0x0}, 0xc00219ba60)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:763 +0xf7
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1()
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:728 +0x415
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:639 +0x273
```
The issue seems to be fixed on master, so let me figure out if something needs to be backported.

--- a/pkg/sql/logictest/testdata/logic_test/distsql_stats
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_stats
@@ -2390,3 +2390,22 @@ CREATE INDEX ON xy (y) WHERE y > 5;
 
 statement error pq: table xy does not contain a non-partial forward index with y as a prefix column
 CREATE STATISTICS xy_partial_idx ON y FROM xy USING EXTREMES;
+
+# Regression test for #100909. Ensure enum is hydrated in SHOW HISTOGRAM.
+statement ok
+CREATE TYPE enum1 as ENUM ('hello', 'hi');
+CREATE TABLE t100909 (x int, y enum1);
+INSERT INTO t100909 VALUES (1, 'hello'), (2, 'hello'), (3, 'hi');
+
+statement ok
+CREATE STATISTICS s1 ON y FROM t100909;
+
+let $hist_id_1
+SELECT histogram_id FROM [SHOW STATISTICS FOR TABLE t100909] WHERE statistics_name = 's1'
+
+query TIRI colnames
+SHOW HISTOGRAM $hist_id_1
+----
+upper_bound  range_rows  distinct_range_rows  equal_rows
+'hello'      0           0                    2
+'hi'         0           0                    1

--- a/pkg/sql/show_histogram.go
+++ b/pkg/sql/show_histogram.go
@@ -16,6 +16,8 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catenumpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/typedesc"
 	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
@@ -74,6 +76,10 @@ func (p *planner) ShowHistogram(ctx context.Context, n *tree.ShowHistogram) (pla
 			}
 
 			v := p.newContainerValuesNode(showHistogramColumns, 0)
+			resolver := descs.NewDistSQLTypeResolver(p.descCollection, p.InternalSQLTxn().KV())
+			if err := typedesc.EnsureTypeIsHydrated(ctx, histogram.ColumnType, &resolver); err != nil {
+				return nil, err
+			}
 			for _, b := range histogram.Buckets {
 				ed, _, err := rowenc.EncDatumFromBuffer(
 					catenumpb.DatumEncoding_ASCENDING_KEY, b.UpperBound,

